sys/freebsd/dev_crypto.txt

```diff
@@ -0,0 +1,68 @@
+# Copyright 2020 syzkaller project authors. All rights reserved.
+# Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+include <sys/time.h>
+include <opencrypto/cryptodev.h>
+
+resource fd_crypto[fd]
+
+openat$crypto(fd const[AT_FDCWD], file ptr[in, string["/dev/crypto"]], flags flags[open_flags], mode const[0]) fd_crypto
+
+ioctl$CIOCFINDDEV(fd fd_crypto, cmd const[CIOCFINDDEV], arg ptr[in, crypt_find_op])
+ioctl$CIOCGSESSION(fd fd_crypto, cmd const[CIOCGSESSION], arg ptr[in, session_op])
+ioctl$CIOCGSESSION2(fd fd_crypto, cmd const[CIOCGSESSION2], arg ptr[in, session2_op])
+ioctl$CIOCCRYPT(fd fd_crypto, cmd const[CIOCCRYPT], arg ptr[in, crypt_op])
+ioctl$CIOCCRYPTAEAD(fd fd_crypto, cmd const[CIOCCRYPTAEAD], arg ptr[in, crypt_aead])
+ioctl$CIOCFSESSION(fd fd_crypto, cmd const[CIOCFSESSION], arg ptr[in, array[int32]])
+
+crypt_find_op {
+	crid	int32
+	name	array[int8, 32]
+}
+
+session_op {
+	cipher		int32
```

Comment:
Would it make sense to provide some values here?

Reply:
Okay. So you mean to define all the cipher values which can be added here (eg. CRYPTO_AES_CBC)?

Reply:
Yes, and in the other places where some specified values are used normally. See [socket_inet.txt:39](https://github.com/google/syzkaller/blob/master/sys/freebsd/socket_inet.txt#L39) for an example.

Reply:
Okay, so should I pass it as flags? Let me work on this and i will update soon.

Reply:
I'm a bit skeptical that it's necessary to specify individual flags. Can't we just specify a range, CRYPTO_ALGORITHM_MIN - CRYPTO_ALGORITHM_MAX? The syzlang grammar supports it: https://github.com/google/syzkaller/blob/master/docs/syscall_descriptions_syntax.md

Reply:
I saw this after I pushed. I have added flags that seemed fit from cryptodev.c. If the flags are not needed and the range should be added instead, I will push another commit.

Reply:
I think a range makes more sense, otherwise this list of algorithms is just going to become stale over time and require manual work to update. I don't have a strong objection to the current approach in the diff though, we can revisit it later once other feedback has been addressed.

Reply:
I saw the range, but mac and cipher range is common, so will it be given the same for both or there can be a distinction?